packages/nuxt/src/core/plugins/plugin-metadata.ts

```diff
@@ -0,0 +1,181 @@
+import type { CallExpression, Property, SpreadElement } from 'estree'
+import type { Node } from 'estree-walker'
+import { walk } from 'estree-walker'
+import { parse } from 'typescript-estree'
+import { defu } from 'defu'
+import { findExports } from 'mlly'
+import type { Nuxt } from '@nuxt/schema'
+import { createUnplugin } from 'unplugin'
+import MagicString from 'magic-string'
+import { normalize } from 'pathe'
+
+// eslint-disable-next-line import/no-restricted-paths
+import type { ObjectPlugin, PluginMeta } from '#app'
+
+// -50: pre-all (nuxt)
+// -40: custom payload revivers (user)
+// -30: payload reviving (nuxt)
+// -20: pre (user) <-- pre mapped to this
+// -10: default (nuxt)
+// 0: default (user) <-- default behavior
+// +10: post (nuxt)
+// +20: post (user) <-- post mapped to this
+// +30: post-all (nuxt)
+
+export const orderMap: Record<NonNullable<ObjectPlugin['enforce']>, number> = {
+  pre: -20,
+  default: 0,
+  post: 20
+}
+
+const metaCache: Record<string, Omit<PluginMeta, 'enforce'>> = {}
+export function extractMetadata (code: string) {
+  let meta: PluginMeta = {}
+  if (metaCache[code]) {
+    return metaCache[code]
+  }
+  walk(parse(code) as Node, {
+    enter (_node) {
+      if (_node.type !== 'CallExpression' || (_node as CallExpression).callee.type !== 'Identifier') { return }
+      const node = _node as CallExpression & { start: number, end: number }
+      const name = 'name' in node.callee && node.callee.name
+      if (name !== 'defineNuxtPlugin' && name !== 'definePayloadPlugin') { return }
+
+      if (name === 'definePayloadPlugin') {
+        meta.order = -40
```

Comment:
Should we have them as a constant map as well to avoid the usage of magic numbers?